ADDING_A_PLUGIN.md

```diff
@@ -0,0 +1,347 @@
+
+[TensorBoard README]: https://github.com/tensorflow/tensorboard/blob/master/README.md
+[YouTube presentation at the TF Dev Summit]: https://www.youtube.com/watch?v=eBbEDRsCmv4&t=5s
+
+As a plugin developer, there are a few concepts you should be especially familiar with.
+
+
+### TensorFlow summaries
+
+The TensorFlow summary system captures tensors as they flow through the graph, and stores them to event logs on disk using [`tf.FileWriter`]. TensorBoard then reads these event log files into memory to extract [`Summary`] protobufs.
+
+> Note: [Protocol Buffers][protobufs] are sort of like JSON objects, except they have schemas, a compiler that generates object mapping code for multiple languages, and a really efficient binary encoding, as well as text encoding format.
+
+The following three fields of the `Summary` protobuf are the most important:
+
+  - The `tag`, which is a string that is uniquely associated with a certain "stream" of related data.
+  - The [`SummaryMetadata` object], which contains metadata associated with that `tag`, including plugin-specific metadata. You can store your own arbitrary binary encoded protobufs in its fields.
+  - The [`TensorProto`] content field. We use `TensorProto` because they are generally a natural way of representing data in TensorFlow. Since `TensorProtos` support arbitrary byte arrays, you can put any kind of data in them.
+
+[`tf.FileWriter`]: https://www.tensorflow.org/api_docs/python/tf/summary/FileWriter
+[`Summary`]: https://github.com/tensorflow/tensorflow/blob/r1.3/tensorflow/core/framework/summary.proto#L62
+[protobufs]: https://developers.google.com/protocol-buffers/
+[`SummaryMetadata` object]: https://github.com/tensorflow/tensorflow/blob/r1.3/tensorflow/core/framework/summary.proto#L38
+[`TensorProto`]: https://github.com/tensorflow/tensorflow/blob/r1.3/tensorflow/core/framework/tensor.proto#L14
+
+
+### TensorBoard data model: runs and tags
+
+Let's say you train a model with three sets of hyperparameters. Logically, you want to keep track of which data was associated with which set of hyperparameters, or which 'run' of TensorFlow. That gives us the TensorBoard concept of a `run`.
+
+A `run` describes a sequence of events that came from a single execution context of TensorFlow. Each event has a timestamp and a monotonically increasing step counter, which was passed to the [`add_summary`] method.
+
+You'll probably track multiple parameters in your model: say, the accuracy, precision, and cross-entropy. Each of these will have its own `tag`. Each `tag` tracks how a particular metric or measurement evolves over time. For each `run` there will be one or more `tag`.
```

Comment:
Changed to "each run may have many `tag`s". Technically, a run could have zero tags.